bentoml/_internal/bento/build_config.py

```diff
@@ -702,6 +703,54 @@ def to_yaml(self, stream: t.TextIO) -> None:
         raise NotImplementedError
 
 
+@attr.define(frozen=True, init=False)
+class BentoPatternSpec:
+    _include: PathSpec = attr.field(
+        converter=lambda x: PathSpec.from_lines("gitwildmatch", x)
+    )
+    _exclude: PathSpec = attr.field(
+        converter=lambda x: PathSpec.from_lines("gitwildmatch", x)
+    )
+    # we want to ignore .git folder in cases the .git folder is very large.
+    git: PathSpec = attr.field(
+        default=PathSpec.from_lines("gitwildmatch", [".git"]), init=False
+    )
+
+    def __init__(self, build_config: "FilledBentoBuildConfig") -> None:
+        return self.__attrs_init__(build_config.include, build_config.exclude)  # type: ignore (unfinished __attr_init__ type)
```

Comment:
Let's pass in just what we need, the include and exclude fields, to be specific.